### PR TITLE
[Web-25] course search window

### PR DESCRIPTION
**Pull request by emilyPhee**
https://user-images.githubusercontent.com/57051508/212577134-be6588f9-ab18-4eca-a9e3-7a6a81818963.mov

- Add CourseSearchWindow component
  - Implement animation for open/close component with a search courses button

### CODE REVIEW

File: new-nextjs-client/pages/index.tsx [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
 import AddYearDropdown from '@components/index-page/AddYearDropdown/AddYearDropdown';
 import Avatar from '@components/index-page/Avatar/Avatar';
+import AppButton from '@components/shared/AppButton/AppButton';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { search } from '@styles/fontawesome';
+
+import CourseSeachWindow from '@components/CourseSearchWindow/CourseSeachWindow';
 
 export default function Home() {
+  const [toggle, setToggle] = useState(false);
```
Comment (anthopark) [inline new-nextjs-client/pages/index.tsx]: Let's be more specific about naming: "searchWindowToggle"

File: new-nextjs-client/pages/index.tsx [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
-import { ReactNode } from 'react';
+import { ReactNode, useState } from 'react';
 import ThemeToggler from '@components/ThemeToggler/ThemeToggler';
 import AddYearDropdown from '@components/index-page/AddYearDropdown/AddYearDropdown';
 import Avatar from '@components/index-page/Avatar/Avatar';
+import AppButton from '@components/shared/AppButton/AppButton';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { search } from '@styles/fontawesome';
+
+import CourseSeachWindow from '@components/CourseSearchWindow/CourseSeachWindow';
```
Comment (anthopark) [inline new-nextjs-client/pages/index.tsx]: typo: CourseSearchWindow

**Review by emilyPhee — COMMENTED** [review]
reviewed

**Comment by emilyPhee** [comment]
[[Web] React-Spring for CourseSearchWindow](https://trello.com/c/zGSu5qYS/25-web-react-spring-for-coursesearchwindow)

File: new-nextjs-client/src/components/CourseSearchWindow/CourseSeachWindow.module.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+$top-animated-div-mt: 2rem;
+
+.animatedDiv {
+  background-color: $white-1;
```
Comment (anthopark) [inline new-nextjs-client/src/components/CourseSearchWindow/CourseSeachWindow.module.scss]: I think you meant to change `$white` to `$bg-color-2`.

**Review by anthopark — COMMENTED** [review]
Great work! Thanks for working on this and implementing the animation with react-spring!

Just left a few minor comments.

**Review by anthopark — APPROVED** [review]

**Comment by emilyPhee** [comment]
> Great work! Thanks for working on this and implementing the animation with react-spring!
> 
> Just left a few minor comments.

Thank you for your review! ☺️

**Pull request merged**